src/App.js

```diff
@@ -1,75 +1,177 @@
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+  handleChange = ({ target }) => {
+    const { value } = target;
+
+    this.setState({
+      todo: value,
+    });
+  }
+
+  handleSubmit = (todo, currentIndex) => {
+    if (todo) {
+      const todoItem = {
+        todoTitle: todo,
+        id: currentIndex,
+        completed: false,
+      };
+
+      this.setState(prevState => ({
+        ...prevState,
+        todo: '',
+        currentIndex: prevState.currentIndex + 1,
+        todosList: [...prevState.todosList, todoItem],
+        filteredTodosList: [...prevState.todosList, todoItem],
+      }));
+    }
+  };
+
+  handleCheck = (
+    { target }, todoItem, filterField, showActiveTodos, showCompletedTodos
+  ) => {
+    this.setState(prevState => ({
+      todosList: prevState.todosList.map((todo) => {
+        if (todo.id === todoItem.id) {
+          return { ...todo, completed: target.checked };
+        }
+
+        return todo;
+      }),
+      filteredTodosList: prevState.filteredTodosList.map((todo) => {
+        if (todo.id === todoItem.id) {
+          return { ...todo, completed: target.checked };
+        }
+
+        return todo;
+      }),
+    }));
+
+    if (filterField === 'active') {
+      showActiveTodos();
+    } else if (filterField === 'completed') {
+      showCompletedTodos();
+    }
+  }
+
+  handleRemove = (todoItem) => {
+    this.setState(prevState => ({
+      todosList: prevState.todosList.filter((todo) => {
+        if (todo.id === todoItem.id) {
+          return false;
+        }
+
+        return true;
+      }),
+      filteredTodosList: prevState.todosList.filter((todo) => {
+        if (todo.id === todoItem.id) {
+          return false;
+        }
 
-      <footer className="footer" style={{ display: 'block' }}>
-        <span className="todo-count">
-          3 items left
-        </span>
-
-        <ul className="filters">
-          <li>
-            <a href="#/" className="selected">All</a>
-          </li>
-
-          <li>
-            <a href="#/active">Active</a>
-          </li>
-
-          <li>
-            <a href="#/completed">Completed</a>
-          </li>
-        </ul>
-
-        <button
-          type="button"
-          className="clear-completed"
-          style={{ display: 'block' }}
+        return true;
+      }),
+    }));
+  }
+
+  showAllTodos = () => {
+    this.setState(prevState => ({
+      filteredTodosList: prevState.todosList,
+      filterField: 'all',
+    }));
+  }
+
+  showActiveTodos = () => {
+    this.setState(prevState => ({
+      filteredTodosList:
+        prevState.todosList.filter(todo => todo.completed === false),
+      filterField: 'active',
+    }));
+  }
+
+  showCompletedTodos = () => {
```

Comment:
for these two functions would be better to create one and pass different arguments to it